README_tcpdevmem.md

```diff
@@ -0,0 +1,277 @@
+# Neper with TCPDirect run instructions
+
+Table of Contents
+- [Neper with TCPDirect run instructions](#neper-with-tcpdirect-run-instructions)
+  - [TCPDirect CUDA: tcp\_stream within Docker container](#tcpdirect-cuda-tcp_stream-within-docker-container)
+    - [Building your own image for testing](#building-your-own-image-for-testing)
+    - [Override CUDA library directory (DLVM)](#override-cuda-library-directory-dlvm)
+  - [TCPDirect UDMA: Compiling tcp\_stream](#tcpdirect-udma-compiling-tcp_stream)
+  - [Running tcp\_stream](#running-tcp_stream)
+    - [Added flags](#added-flags)
+    - [Running tcp\_stream via `multi_neper.py`](#running-tcp_stream-via-multi_neperpy)
+      - [Example of successful output](#example-of-successful-output)
+    - [Running tcp\_stream directly](#running-tcp_stream-directly)
+
+
+## TCPDirect CUDA: tcp_stream within Docker container
+
+A TCPDirect-Cuda enabled Docker image exists at `gcr.io/a3-tcpd-staging-hostpool/neper`. The `run_neper_container.sh` script makes it easy to run Neper:
+
+
+```
+./run_neper_container.sh
+
+# within the container
+FLOWS=2
+BUF_SIZE=409600
+DEVS=eth1,eth2,eth3,eth4
+DSTS=192.168.1.26,192.168.2.26,192.168.3.26,192.168.4.26
+SRCS=192.168.1.23,192.168.2.23,192.168.3.23,192.168.4.23
+./multi_neper.py --hosts $DSTS \
+  --devices $DEVS --buffer-size $BUF_SIZE \
+  --flows $FLOWS --threads $FLOWS \
+  --src-ips $SRCS --log DEBUG \
+  --q-num $FLOWS --phys-len 2147483648 \
+  --client \
+  --mode cuda
+```
+
+
+
+### Building your own image for testing
+
+
+```
+# within the repo directory
+docker build -t neperc .
```

Comment:
Needs a step to copy the include files, no?

Reply:
Should also include cloning the repo, like below for UDMA.

Reply:
added `cp -r ~/cos-kernel/usr/include/ ./usr/` to the code-block here and in the build UDMA section